### ISSUE TITLE
Toggle - CornerstoneToggle improvements

### ISSUE THREAD

**Issue by afercia** [Yoast/yoast-components#681]
Noticed there's room for a few improvements to the Toggle / CornerstoneToggle component:

functional:
- when the toggle is toggled pressing the spacebar, there's the need to prevent the default behavior otherwise the page will scroll

a11y:
- since the toggle emulates a checkbox but it's not a native `<checkbox>` element, it can't use a `<label>` element. Worth noting `<div>` elements are not [labelable elements](https://www.w3.org/TR/html5/sec-forms.html#labelable-element)
- the visible text should be just text, referenced by `aria-labelledby` on the element with role=checkbox
- `aria-label` should be removed, as we want the toggle to always have a visible text

misc:
- the visible text shouldn't have a period at the end
- the `id` is hardcoded, so just one `CornerstoneToggle` can be used in a page

**Comment by afercia** [comment]
On a side note: apologies, I should have checked the a11y part a bit earlier.

**Comment by afercia** [comment]
About the On / Off text position, worth considering translations could make the strings longer and potentially break the layout. This is one of the reasons why in Gutenberg the option of placing On / Off inline with the toggle hasn't got much consensus.

![screen shot 2018-08-01 at 12 54 03](https://user-images.githubusercontent.com/1682452/43520616-99573db4-9593-11e8-8b47-9ba635ee60a9.png)

On MyYoast, we're placing the On / Off text _underneath_ the toggle. which helps a bit.

![screen shot 2018-08-01 at 13 55 31](https://user-images.githubusercontent.com/1682452/43520617-99760e74-9593-11e8-8fb0-c0409b49b93c.png)

I seem to recall a conversation with @hedgefield about this issue, and we considered to try the On / Off text underneath the visible text, something like:

<img width="348" alt="screen shot 2018-08-01 at 14 06 50" src="https://user-images.githubusercontent.com/1682452/43520778-2f1dab94-9594-11e8-87a9-7a57c360bfc8.png">

**Comment by afercia** [comment]
Couple more things:
- when `disable={ true }`, an `aria-disabled="true"` attribute should be rendered to communicate the disabled state
- when `disable={ true }`
  - when clicking: it work correctly and only the `onToggleDisabled` callback is executed
  - when pressing Space bar instead, it still executes the `onSetToggleState` callback, i.e. as it wasn't disabled

On a side note, I'm not sure we need a different callback when it's disabled, as this was something implemented specifically for MyYoast.

**Comment by hedgefield** [comment]
@afercia we did indeed discuss using the gutenberg toggle element, but when we decided to go with our own toggle I didn't consider to try a mix of the two in that way.

First though, I wonder if we can't gain some space by changing the label to `Mark as cornerstone content`, what do you think? If that is sufficient I'd prefer the toggle label to stay where it is.

**Comment by afercia** [comment]
We can try 🙂

**Comment by afercia** [comment]
Hm I guess it should support RTL languages too... 😬 not working well with RTL at the moment.
